test/connectivity/NetworkController/LinuxFirewallCommands.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace NetworkController
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class LinuxFirewallCommands : IFirewallCommands
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<LinuxFirewallCommands>();
+        readonly string networkInterfaceName;
+
+        public LinuxFirewallCommands(string networkInterfaceName)
+        {
+            this.networkInterfaceName = networkInterfaceName;
+        }
+
+        public async Task SetAcceptRule(CancellationToken cs)
+        {
+            var exitCode = await CommandExecutor.ExecuteCommand(
+                "tc",
+                $"qdisc delete dev {this.networkInterfaceName} root netem loss 100%",
+                cs);
+
+            var exitMessage = exitCode == 0 ? "Removed" : $"Exit code {exitCode} failed to remove";
```

Comment:
Should we throw an exception if this fails?

Reply:
I changed to return bool, I'll change it again when adding the implementation for reporting to the analyzer, maybe report the actual error code

---
In reply to: [353014050](https://github.com/Azure/iotedge/pull/2040#discussion_r353014050) [](ancestors = 353014050)